rtos/source/ThisThread.cpp

```diff
@@ -251,4 +251,32 @@ const char *get_name()
 #endif
 }
 
+osPriority_t ThisThread::get_priority()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osPriorityError;
+    }
+
+    return static_cast<osPriority_t>(osThreadGetPriority(id));
+#else
+    return static_cast<osPriority_t>(osPriorityError);
+#endif
+}
+
+osStatus_t ThisThread::suspend()
+{
+#if MBED_CONF_RTOS_PRESENT
+    osThreadId_t id = osThreadGetId();
+    if (id == nullptr) {
+        return osError;
+    }
+
+    return static_cast<osStatus_t>(osThreadSuspend(id));
+#else
+    return static_cast<osStatus_t>(osError);
```

Comment:
Suspend on baremetal might be better off being a no-op, rather than an error.

The RTOS API for baremetal is about application portability, so folks can run their existing apps. If we make their existing apps start getting errors where things worked before, we unfortunately break portability.

Reply:
I could pass `osOK` = 0, if that would suffice?

Reply:
Let's improve these in https://github.com/ARMmbed/mbed-os/pull/12388

I'll accept the code as-is into the feature branch.